pkg/adaptor/proxy/proxy.go

```diff
@@ -97,7 +103,13 @@ func (p *agentProxy) dial(ctx context.Context, address string) (net.Conn, error)
 		// Since it's not possible to know the IP address of the pod VM apriori,
 		// we are using a well-defined hostname here. Other option is to create
 		// certificates with IP SAN having all the IPs in the network range
-		config.ServerName = podvmServername
+		// When CA service is enabled, a server certificate is automatically generated for
+		// the instance VM name.
+		if p.caService != nil {
+			config.ServerName = p.serverName
+		} else {
+			config.ServerName = podvmServername
+		}
 
```

Comment:
With automatic mTLS option available, do we really need to keep non-TLS option ?

Reply:
A possible use case is debugging the proxy communication to by analyzing packet dumps captured tcpdump. I agree that non-TLS option is not necessary in most cases.

Reply:
Right, for debugging we might need plaintext. Thanks.